src/sagemaker/local/image.py

```diff
@@ -139,7 +104,15 @@ def train(self, input_data_config, hyperparameters):
         compose_command = self._compose()
 
         _ecr_login_if_needed(self.sagemaker_session.boto_session, self.image)
-        _execute_and_stream_output(compose_command)
+        process = subprocess.Popen(compose_command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+        try:
+            _stream_output(process)
```

Comment:
Do you need this.

Doesn't something like this work:

```python
>>> subprocess.check_call(["ls", "asdf"], stderr=sys.stdout.fileno())
ls: asdf: No such file or directory
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/Cellar/python/2.7.14/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 186, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['ls', 'asdf']' returned non-zero exit status 1
```

I think the interface is a bit different in python 3, but we can work through this.

Reply:
Talked IRL, this won't work in Jupyter because Jupyter replaces the stderr object.